src/Models/LoginModel.php

```diff
@@ -56,6 +56,12 @@ public function recordLoginAttempt(
         ?string $userAgent = null,
         $userId = null
     ): void {
+        $this->disableDBDebug();
+
+        if ($this->db->getPlatform() === 'OCI8' && $identifier === '') {
+            $identifier = ' ';
```

Comment:
... and this?

Reply:
Ah I see the whole commit message now, that explains this one. Seems like this is probably a bug in the framework driver itself, right?

Reply:
I'm not sure.
Oracle sees `''` as `NULL`.

Reply:
Oracle Database currently treats a character value with a length of zero as null. However, this may not continue to be true in future releases, and Oracle recommends that you do not treat empty strings the same as nulls.

More [see](https://docs.oracle.com/cd/B19306_01/server.102/b14200/sql_elements005.htm).

Reply:
@datamweb I knew it, but I don't know what should we do in this case.
We try to insert `''`, but Oracle treats it as `NULL` and says "cannot insert NULL"

Reply:
@kenjis  I'm not sure if it's helpful, but take a look if you have time.
https://stackoverflow.com/questions/2185743/is-it-possible-to-store-empty-string-as-a-non-null-value-in-the-database/73396113#73396113

Reply:
> Use an alternate representation, such as a single space character, for an empty string.

It is what I did here.

Reply:
@ytetsuro have any recommendations? Or @sclubricants?